packages/plugin-npm/sources/index.ts

```diff
@@ -12,6 +12,21 @@ const plugin: Plugin = {
       type: SettingsType.STRING,
       default: `https://registry.yarnpkg.com`,
     },
+    npmAlwaysAuth: {
+      description: `URL of the selected npm registry (note: npm enterprise isn't supported)`,
+      type: SettingsType.BOOLEAN,
+      default: false,
+    },
+    npmAuthIdent: {
+      description: `Authentication identity for the npm registry (_auth in npm and yarn v1)`,
+      type: SettingsType.SECRET,
+      default: null,
+    },
+    npmAuthToken: {
```

Comment:
Thinking about it, I wonder if maybe we should also have something like this (I remember that different scopes may target different registries):

```yaml
npmScopes:
  my-scope:
    npmRegistryServer: https://localhost
    npmAuthIdent: foo
```

But we can do that a follow up, just to keep in mind in the design.

Reply:
Current configuration goes one step further: it allows setting authentication configuration per registry as well (.npmrc allows that for sure, I don't know about .yarnrc) That's actually what `npm login` generates: a `_authToken` linked to the registry you used to log in.

So our config could be

```yaml
npmScopes:
  my-scope:
    npmRegistryServer: https://localhost

npmRegistries:
  //localhost:
    npmAuthIdent: foo
```

although the snippet you posted would have the same result.

Reply:
I see - that could be useful when multiple scopes have the same registry, I guess 🤔 ok then!

Reply:
I'd like to postpone this to another PR if that's fine by you, I generally prefer multiple smaller PRs with limited scope over larger PRs containing many changes.

Reply:
Of course 👍